src/analysis/special_functions/trigonometric/chebyshev.lean

```diff
@@ -4,24 +4,52 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johan Commelin
 -/
 import analysis.complex.basic
-import ring_theory.polynomial.chebyshev
 import data.complex.exponential
+import data.polynomial.algebra_map
+import ring_theory.polynomial.chebyshev
 
 /-!
 # Multiple angle formulas in terms of Chebyshev polynomials
 
-* `polynomial.chebyshev.T_complex_cos`: the `n`-th Chebyshev polynomial evaluates on `complex.cos θ`
-  to the value `complex.cos (n * θ)`.
+This file gives the trigonometric characterizations of Chebyshev polynomials, for both the real and
+complex cosine.
 -/
 
 namespace polynomial.chebyshev
+open polynomial
+
+variables {R A : Type*} [comm_ring R] [comm_ring A] [algebra R A]
+
+@[simp] lemma aeval_T (x : A) (n : ℕ) : aeval x (T R n) = (T A n).eval x :=
+by rw [aeval_def, eval₂_eq_eval_map, map_T]
+
+@[simp] lemma aeval_U (x : A) (n : ℕ) : aeval x (U R n) = (U A n).eval x :=
+by rw [aeval_def, eval₂_eq_eval_map, map_U]
+
+@[simp] lemma algebra_map_eval_T (x : R) (n : ℕ) :
+  algebra_map R A ((T R n).eval x) = (T A n).eval (algebra_map R A x) :=
+by rw [←aeval_algebra_map_apply, aeval_T]
+
+@[simp] lemma algebra_map_eval_U (x : R) (n : ℕ) :
+  algebra_map R A ((U R n).eval x) = (U A n).eval (algebra_map R A x) :=
+by rw [←aeval_algebra_map_apply, aeval_U]
+
+@[simp, norm_cast] lemma of_real_eval_T : ∀ x n, ((T ℝ n).eval x : ℂ) = (T ℂ n).eval x :=
+@algebra_map_eval_T ℝ ℂ _ _ _
+
+@[simp, norm_cast] lemma of_real_eval_U : ∀ x n, ((U ℝ n).eval x : ℂ) = (U ℂ n).eval x :=
+@algebra_map_eval_U ℝ ℂ _ _ _
```

Comment:
```suggestion
@[simp, norm_cast] lemma complex_of_real_eval_T : ∀ x n, ((T ℝ n).eval x : ℂ) = (T ℂ n).eval x :=
@algebra_map_eval_T ℝ ℂ _ _ _

@[simp, norm_cast] lemma complex_of_real_eval_U : ∀ x n, ((U ℝ n).eval x : ℂ) = (U ℂ n).eval x :=
@algebra_map_eval_U ℝ ℂ _ _ _
```
since this could also refer to `ennreal.of_real` etc.